components/scream/cmake/ScreamUtils.cmake

```diff
@@ -44,90 +44,57 @@ function(CreateUnitTest target_name target_srcs scream_libs)
-  #       or you would have either a) a keyword followed by nothing (error),
-  #       or b) to add a space at front or back (causing EKAT to think that
-  #       that option is present, when it's not).
-  set (cut_options EXCLUDE_TEST_SESSION)
-  if (CreateUnitTest_EXCLUDE_MAIN_CPP)
-    list(APPEND cut_options EXCLUDE_MAIN_CPP)
-  endif()
-  if (CreateUnitTest_SERIAL)
-    list(APPEND cut_options SERIAL)
-  endif()
-  if (CreateUnitTest_PRINT_OMP_AFFINITY)
-    list(APPEND cut_options PRINT_OMP_AFFINITY)
-  endif()
-  if (CreateUnitTest_DEP)
-    list(APPEND cut_options DEP ${CreateUnitTest_DEP})
-  endif()
-  if (CreateUnitTest_PROPERTIES)
-    list(APPEND cut_options PROPERTIES ${CreateUnitTest_PROPERTIES})
-  endif()
-  if (CreateUnitTest_LABELS)
-    list(APPEND cut_options LABELS ${CreateUnitTest_LABELS})
-  endif()
-  if (CreateUnitTest_MPI_RANKS)
-    list(APPEND cut_options MPI_RANKS ${CreateUnitTest_MPI_RANKS})
-  endif ()
-  if (CreateUnitTest_THREADS)
-    list(APPEND cut_options THREADS ${CreateUnitTest_THREADS})
-  endif ()
-  if (CreateUnitTest_EXE_ARGS)
-    list(APPEND cut_options EXE_ARGS ${CreateUnitTest_EXE_ARGS})
-  endif ()
+  #
+  # Convert CreateUnitTest options/args to EkatCreateUnitTest options/args
+  #
+  set(skip_list "INCLUDE_DIRS;COMPILER_F_FLAGS")
+
+  set(cut_options EXCLUDE_TEST_SESSION)
+
+  foreach(item IN LISTS options oneValueArgs multiValueArgs)
+    if (NOT item IN_LIST skip_list AND CreateUnitTest_${item})
+      if (item IN_LIST options)
+        list(APPEND cut_options ${item})
+      else()
+        list(APPEND cut_options ${item} ${CreateUnitTest_${item}})
+      endif()
+    endif()
+  endforeach()
+
   if (SCREAM_MPI_EXTRA_ARGS)
     list(APPEND cut_options MPI_EXTRA_ARGS ${SCREAM_MPI_EXTRA_ARGS})
   endif ()
-  if (CreateUnitTest_COMPILER_DEFS)
-    list(APPEND cut_options COMPILER_DEFS ${CreateUnitTest_COMPILER_DEFS})
-  endif ()
-  if (CreateUnitTest_COMPILER_C_DEFS)
-    list(APPEND cut_options COMPILER_C_DEFS ${CreateUnitTest_COMPILER_C_DEFS})
-  endif ()
-  if (CreateUnitTest_COMPILER_CXX_DEFS)
-    list(APPEND cut_options COMPILER_CXX_DEFS ${CreateUnitTest_COMPILER_CXX_DEFS})
-  endif ()
-  if (CreateUnitTest_COMPILER_F_DEFS)
-    list(APPEND cut_options COMPILER_F_DEFS ${CreateUnitTest_C_COMPILER_DEFS})
-  endif ()
-  if (CreateUnitTest_COMPILER_FLAGS)
-    list(APPEND cut_options COMPILER_FLAGS ${CreateUnitTest_C_COMPILER_DEFS})
-  endif ()
-  if (CreateUnitTest_COMPILER_C_FLAGS)
-    list(APPEND cut_options COMPILER_C_FLAGS ${CreateUnitTest_C_COMPILER_DEFS})
-  endif ()
-  if (CreateUnitTest_COMPILER_CXX_FLAGS)
-    list(APPEND cut_options COMPILER_CXX_FLAGS ${CreateUnitTest_C_COMPILER_DEFS})
-  endif ()
+
   if (CreateUnitTest_COMPILER_F_FLAGS OR SCREAM_Fortran_FLAGS)
-    list(APPEND cut_options COMPILER_F_FLAGS ${CreateUnitTest_C_COMPILER_DEFS} ${SCREAM_Fortran_FLAGS})
+    list(APPEND cut_options COMPILER_F_FLAGS ${CreateUnitTest_F_COMPILER_DEFS} ${SCREAM_Fortran_FLAGS})
```

Comment:
Also, I was a bit confused why Fortran was the only language that got SCREAM_Fortran_FLAGS included.